File: educational-core/src/com/jetbrains/edu/learning/settings/OptionsProvider.java
package com.jetbrains.edu.learning.settings;

import com.intellij.openapi.extensions.ExtensionPointName;
import com.intellij.openapi.options.Configurable;
import com.intellij.openapi.options.ConfigurableEP;

public interface OptionsProvider extends Configurable {
  ExtensionPointName<ConfigurableEP<OptionsProvider>> EP_NAME = ExtensionPointName.create("Educational.optionsProvider");
}

File: intellij-plugin/educational-core/src/com/jetbrains/edu/learning/settings/OptionsProvider.kt
package com.jetbrains.edu.learning.settings

import com.intellij.openapi.extensions.ExtensionPointName
import com.intellij.openapi.options.Configurable
import com.intellij.openapi.options.ConfigurableEP

interface OptionsProvider : Configurable {
  fun isAvailable(): Boolean = true

  companion object {
    val EP_NAME: ExtensionPointName<ConfigurableEP<OptionsProvider>> = ExtensionPointName.create("Educational.optionsProvider")
  }
}
